Snapshot to GCF

```
ERROR: (gcloud.functions.deploy) OperationError: code=3, message=Function failed on loading user code. This is likely due to a bug in the user code. Error message: Picked up JAVA_TOOL_OPTIONS: -XX:MaxRAM=256m -XX:MaxRAMPercentage=70
2023-05-03 22:00:58.791:INFO::main: Logging initialized @579ms to org.eclipse.jetty.util.log.StdErrLog
Error: Exception in thread "main" java.lang.UnsupportedClassVersionError: io/micronaut/gcp/function/http/HttpFunction has been compiled by a more recent version of the Java Runtime (class file version 61.0), this version of the Java Runtime only recognizes class file versions up to 55.0
	at java.base/java.lang.ClassLoader.defineClass1(Native Method)
	at java.base/java.lang.ClassLoader.defineClass(ClassLoader.java:1017)
	at java.base/java.security.SecureClassLoader.defineClass(SecureClassLoader.java:174)
	at java.base/java.net.URLClassLoader.defineClass(URLClassLoader.java:555)
	at java.base/java.net.URLClassLoader$1.run(URLClassLoader.java:458)
	at java.base/java.net.URLClassLoader$1.run(URLClassLoader.java:452)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.net.URLClassLoader.findClass(URLClassLoader.java:451)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:589)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:522)
	at com.google.cloud.functions.invoker.runner.Invoker.loadFunctionClass(Invoker.java:272)
	at com.google.cloud.functions.invoker.runner.Invoker.startServer(Invoker.java:235)
	at com.google.cloud.functions.invoker.runner.Invoker.main(Invoker.java:121)
. Please visit https://cloud.google.com/functions/docs/troubleshooting for in-depth troubleshooting documentation.
Error: Process completed with exit code 1.
```

--- a/.github/workflows/gcf-snapshot.yml
+++ b/.github/workflows/gcf-snapshot.yml
@@ -31,4 +31,4 @@ jobs:
       - name: Deploy to Cloud Function
         run: |
           cd starter-gcp-function/build/libs
-          gcloud functions deploy ${{ secrets.GCLOUD_SNAPSHOT_FUNCTION_NAME }} --trigger-http --project ${{ secrets.GCLOUD_FUNCTION_PROJECT }} --region  us-central1 --allow-unauthenticated --service-account=${{ secrets.GCLOUD_EMAIL }} --quiet --entry-point io.micronaut.gcp.function.http.HttpFunction --runtime java11
+          gcloud functions deploy ${{ secrets.GCLOUD_SNAPSHOT_FUNCTION_NAME }} --trigger-http --project ${{ secrets.GCLOUD_FUNCTION_PROJECT }} --region  us-central1 --allow-unauthenticated --service-account=${{ secrets.GCLOUD_EMAIL }} --quiet --entry-point io.micronaut.gcp.function.http.HttpFunction --runtime java17

